enterprise/api/v1/views.py

```diff
@@ -289,14 +392,9 @@ def retrieve(self, request, pk=None):  # pylint: disable=invalid-name
         Returns:
             (Response): DRF response object containing course catalogs.
         """
-        # fetch course catalog for the given catalog id.
         catalog_api = CourseCatalogApiClient(request.user)
         catalog = catalog_api.get_catalog(pk)
-
-        if not catalog:
-            logger.error("Unable to fetch API response for given catalog from endpoint '/catalog/%s/'.", pk)
-            raise NotFound("The resource you are looking for does not exist.")
-
+        self.ensure_data_exists(request, catalog)
```

Comment:
@UmanShahzad Let's preserve the custom error message here (pass it as `error_message`).